core/logger/Logger.cpp

```diff
@@ -305,7 +320,11 @@ void Logger::LoadConfig(const std::string& filePath) {
         spdlog::register_logger(logger);
         logger->set_level(loggerCfg.level);
         logger->set_pattern(DEFAULT_PATTERN);
-        logger->flush_on(loggerCfg.level);
+        if (name == "/apsara/sls/ilogtail" && aliyun_logtail_log_level != "" && envLogLevel) {
```

Comment:
&& aliyun_logtail_log_level != "" && envLogLevel should be && !aliyun_logtail_log_level.empty()
because 0 means trace. it's valid.